tools/clang/unittests/HLSL/OptionsTest.cpp

```diff
@@ -380,3 +384,233 @@ TEST_F(OptionsTest, TestPreprocessOption) {
+
+  int retVal =
+      ReadDxcOpts(optionTable, DxcFlags, argStrings, dxcOpts, errorStream);
+  EXPECT_EQ(retVal, 0);
+  errorStream.flush();
+  EXPECT_EQ(errorString.empty(), true);
+  EXPECT_EQ(
+      static_cast<uint32_t>(hlsl::options::ComputeSerializeDxilFlags(dxcOpts)),
+      ExpectFlags);
+}
+
+static uint32_t CombineFlags(llvm::ArrayRef<SerializeDxilFlags> flags) {
+  uint32_t result = 0;
+  for (SerializeDxilFlags f : flags)
+    result |= static_cast<uint32_t>(f);
+  return result;
+}
+
+struct SerializeDxilFlagsTest {
+  const char *command;
+  uint32_t flags;
+};
+
+using F = SerializeDxilFlags;
+
+TEST_F(OptionsTest, SerializeDxilFlags) {
+  // Test cases for SerializeDxilFlags
+  // These cases are generated by group flags and do a full combination.
+  // [("-Qstrip_rootsignature", {"F::StripRootSignature"}),\
+      // ("", set())]
+  // [("-Qkeep_reflect_in_dxil", set()),\
+      // ("", {"F::StripReflectionFromDxilPart"})]
+  // [("-Qstrip_reflect", {}), \
+    // ("", {"F::IncludeReflectionPart"})]
+  // [("-Zss -Zs", {"F::IncludeDebugNamePart","F::DebugNameDependOnSource"}),\
+      // ("-Zsb", {"F::IncludeDebugNamePart"}), \
+      // ("-FdDbgName.pdb", {"F::IncludeDebugNamePart"}), \
+      // ("-Zi", {"F::IncludeDebugNamePart"}), \
+      // ("-Zsb -Qembed_debug -Zi", {"F::IncludeDebugInfoPart","F::IncludeDebugNamePart"}), \
+      // ("-FdDbgName.pdb -Qembed_debug -Zi", {"F::IncludeDebugInfoPart","F::IncludeDebugNamePart"}), \
+      // ("", set())]
+
+  SerializeDxilFlagsTest Tests[] = {
+    {"-Qstrip_rootsignature -Qkeep_reflect_in_dxil -Qstrip_reflect -Zss -Zs",
+    CombineFlags({F::IncludeDebugNamePart, F::DebugNameDependOnSource, F::StripRootSignature})
+    },
+    {"-Qstrip_rootsignature -Qkeep_reflect_in_dxil -Qstrip_reflect -Zsb",
+    CombineFlags({F::IncludeDebugNamePart, F::StripRootSignature})
+    },
+    {"-Qstrip_rootsignature -Qkeep_reflect_in_dxil -Qstrip_reflect -FdDbgName.pdb",
+    CombineFlags({F::IncludeDebugNamePart, F::StripRootSignature})
+    },
+    {"-Qstrip_rootsignature -Qkeep_reflect_in_dxil -Qstrip_reflect -Zi",
+    CombineFlags({F::IncludeDebugNamePart, F::StripRootSignature})
+    },
+    {"-Qstrip_rootsignature -Qkeep_reflect_in_dxil -Qstrip_reflect -Zsb -Qembed_debug -Zi",
+    CombineFlags({F::IncludeDebugNamePart, F::IncludeDebugInfoPart, F::StripRootSignature})
+    },
+    {"-Qstrip_rootsignature -Qkeep_reflect_in_dxil -Qstrip_reflect -FdDbgName.pdb -Qembed_debug -Zi",
+    CombineFlags({F::IncludeDebugNamePart, F::IncludeDebugInfoPart, F::StripRootSignature})
+    },
+    {"-Qstrip_rootsignature -Qkeep_reflect_in_dxil -Qstrip_reflect ",
+    CombineFlags({F::StripRootSignature})
+    },
+    {"-Qstrip_rootsignature -Qkeep_reflect_in_dxil  -Zss -Zs",
+    CombineFlags({F::IncludeDebugNamePart, F::IncludeReflectionPart, F::DebugNameDependOnSource, F::StripRootSignature})
+    },
+    {"-Qstrip_rootsignature -Qkeep_reflect_in_dxil  -Zsb",
```

Comment:
Since we are imminently going to be clang-formatting things, it would be nice to clang-format the new code coming in.